pkg/cloudprovider/aws/instancetype.go

```diff
@@ -70,6 +72,10 @@ func (i *InstanceType) Memory() *resource.Quantity {
 }
 
 func (i *InstanceType) Pods() *resource.Quantity {
```

Comment:
I think we need to update the Overhead logic to always use the ENILimited MaxPods value, even if not using max pods. We're gonna need a nice verbose comment on this including the fact that we want to make it more accurate XD.

Reply:
Ok. So... are you proposing we don't make any changes (minus a comment update) in this func then since that's what it's currently doing/using?

Reply:
Sorry if this is a bit confusing. 

The Pods() function should return 
- 110 if not eni limited
- the limit if eni limited

The Overhead() function should return
- overhead calculated by the eni limited pod count, regardless of the result of Pods().

I think you're going to minimally need a help to extract it out. I think this was outlined in slack, but it's pretty confusing. The idea was to just follow the eksoptimizedami convention. @suket22 may be the best person for this.

Reply:
Ah! Gotcha... I haven't spent enough time in the scheduling part, so had totally missed the `Overhead()` function. I can take a stab at it and get feedback on it.

Reply:
Alrighty... ready for another look. I added some comments, but nothing too crazy. The blame can take you back to this PR with more history too. Feel free to add more if you'd like 👍